faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -387,10 +389,16 @@ public void syncLastTxnTime(long newTxnTime) {
    *
    * @param path the relative path of the resource.
    * @param queryTimeout the query timeout for the current request.
+   * @param traceId A unique identifier for this query. Adheres to the
+   *                [W3C Trace Context](https://w3c.github.io/trace-context/) spec.
+   * @param tags Key-value pair metadata to associate with this query.
    * @return a {@link CompletableFuture} containing the HTTP Response.
    */
-  public CompletableFuture<HttpResponse<String>> get(String path, Optional<Duration> queryTimeout) {
-    return performRequest("GET", path, Optional.empty(), Map.of(), queryTimeout);
+  public CompletableFuture<HttpResponse<String>> get(String path,
+                                                     Optional<Duration> queryTimeout,
+                                                     Optional<String> traceId,
+                                                     Optional<Map<String, String>> tags) {
```

Comment:
might be worth making a Tag class that can be constructed if and only if the underlying spec meets the requirements for tag keys and tag values. Customers might find that helpful.

You could then change this to be a Set.

For collections, you probably don't need to make it `Optional` as you can just use an empty Collection for not providing it.

e.g.

```(Java)
package the.route.models.tags;

public final class Tag {
    private final String key;
    private final String value;

    public TagKey(String key, String value) {
        validate(REGEX, value); // throws if invalid
        validate(REGEX, key); // throws if invalid
        this.key = key;
        this.value = value;
    }

   // implement hashCode / equals
}

Reply:
Agreed, added a class to encapsulate

Reply:
I should add, while the default, empty Collection is totally doable in Scala, that won't work in the Java driver - hence the Optional. Which, annoyingly, still needs to get cast everywhere, but that's a different issue.